fix(transformer/typescript): unexpectedly removed class binding from ExportNamedDeclaration

Dunqing opened the pull request:
The original `SymbolFlags` methods were a bit confusing I renamed and re-implemented them.

Dunqing commented:
* **#4351** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4351?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Boshen commented on tasks/transform_conformance/tests/babel-plugin-transform-typescript/test/fixtures/class-export/input.js:
```diff
@@ -0,0 +1,2 @@
+class Foo {}
+export { Foo };
```
Should we have a test including all declarations? `class`, `function`, `interface` etc

Dunqing replied:
Done

Dunqing committed:
fix(transformer/typescript): unexpectedly removed class binding from ExportNamedDeclaration (#4351)

The original `SymbolFlags` methods were a bit confusing I renamed and re-implemented them.
